### DISCUSSION TITLE
Combine two images (Photo + Depth Map) into one HEIC file as iOS Portrait Photo

### DISCUSSION THREAD

**Discussion by DeanPix** [ImageMagick/ImageMagick#3892] [Help]
Hi,
Today I have a completely opposite question. Many want to bring out the depth map from a portrait photo taken with an iPhone.
And I would like to "create" a portrait photo made of my own layers, that is a color photo and my own depth map.

![depthmap](https://user-images.githubusercontent.com/67162951/124138059-94b08000-da86-11eb-8663-6e1eba97fb97.jpg)

I would like this HEIC photo to be seen in iPhone as Portrait Mode like in the attached example.

![portrait_ipgone](https://user-images.githubusercontent.com/67162951/124138108-a003ab80-da86-11eb-837f-6f2546abcb7e.jpg)

Many third-party iPhone applications (example: Focos) have the ability to create a depth map from a regular photo and save it as a portrait photo so that the iPhone sees the photo as if it were taken in this portrait mode. Unfortunately, none of these applications allows you to import your own depth map, it can only generate it from the photo itself (which is often very bad).
Hope you know what I mean. You are the last resort. 
I was looking on the web for various applications, whether on MacOS or iPhone that can do that, but to no avail.

Thanks,
Marcin

**Comment by snibgo** [comment]
> I would like this HEIC photo to be seen in iPhone as Portrait Mode ...

> ... and save it as a portrait photo so that the iPhone sees the photo as if it were taken in this portrait mode.

What in the photo file defines it as "Portrait Mode"? Perhaps it is some metadata, and perhaps that can be set with exiftool or exif2.

I suggest you link to a file that is "Portrait Mode", and a similar file that is not "Portrait Mode". Perhaps we can find the relevant difference.

**Reply by DeanPix** [reply]
It turns out that not only HEIC can be seen as a portrait. The JPG file also includes this information along with the depth map.
There are 2 examples at this link, a normal photo and a portrait photo taken with the iPhone 12 Pro Max.

https://www.dropbox.com/sh/pcdr0acetkscgjg/AADTA7AaN9eL-wwNker5Pe21a?dl=0

**Comment by snibgo** [comment]
I can't see anything obvious in exiftool of those files.

A internet search doesn't pin down the difference, but some pages suggest that "portrait mode" jpegs come wth an extra file. For example: https://forums.macrumors.com/threads/workflow-from-iphone-to-mac-heic-container-questions.2219084/

> When you take a Portrait Mode photo as JPG and then view it in Image Capture, you see two files: IMG_1234.jpg and IMG_E1234.jpg. If you import just IMG_1234.jpg, you get IMG_1234.jpg and IMG_1234.aae. IMG_1234.jpg does not contain the Portrait Mode effect in itself, but that info is stored in the .aae file. Importing both of these files into Photos does make the image show with the Portrait Mode effect in Photos, which can be toggled on and off, edited, etc.

So, does your "portrait" photo have an extra jpg file with an extra "E", such as IMG_E1234.jpg? Or an AAE file such as IMG_1234.aae?

**Comment by Michael-Z-Freeman** [comment]
Hi. I would also like to do this. Is ImageMagick capable of adding the depth map auxiliary image to an existing HEIC photo ? Cheers.
